src/IonTimestamp.ts

```diff
@@ -310,154 +281,173 @@ export class Timestamp {
-    return this.getPrecision() === expected.getPrecision() && this.offset === expected.offset && this.dataModelEquals(expected);
-  }
-
-  dataModelEquals(expected : Timestamp) : boolean {
-      switch (this.precision) {
-          case Precision.NULL:
-              return expected.precision === Precision.NULL;
-          case Precision.SECONDS:
-              if(!this.seconds.equals(expected.seconds)) return false;
-          case Precision.HOUR_AND_MINUTE:
-              if(this.minute !== expected.minute || this.hour !== expected.hour) return false;
-          case Precision.DAY:
-              if(this.day !== expected.day) return false;
-          case Precision.MONTH:
-              if(this.month !== expected.month) return false;
-          case Precision.YEAR:
-              if(this.year !== expected.year) return false;
-      }
-      return true;
-  }
-
-  stringValue() : string {
-    let image: string;
-    let t: Timestamp = this;
-
-    switch (t.precision) {
-      default: throw { msg: "invalid value for timestamp precision", where: "IonValueSupport.timestamp.toString" };
-      case Precision.NULL:
-        return "null.timestamp";
-      case Precision.SECONDS:
-        //formats decimal to timestamp second, adds a leading 0 and/or cuts off the trailing period
-        image = t.seconds.toString();
-        if(image.charAt(1)  === '.') image = "0" + image;
-        if(image.charAt(image.length - 1) === '.') image = image.slice(0, image.length - 1);
-      case Precision.HOUR_AND_MINUTE:
-        image = _to_2_digits(t.minute) + (image ? ":" + image : "");
-        image = _to_2_digits(t.hour) + (image ? ":" + image : "");
-      case Precision.DAY:
-        image = _to_2_digits(t.day) + (image ? "T" + image : "T");
-      case Precision.MONTH:
-        image = _to_2_digits(t.month) + (image ? "-" + image : "");
-      case Precision.YEAR:
-        if (t.precision === Precision.YEAR) {
-          image = to_4_digits(t.year) + "T";
-        } else if (t.precision === Precision.MONTH) {
-          image = to_4_digits(t.year) + "-" + image + "T";
-        } else {
-          image = to_4_digits(t.year) + "-" + image;
+
+    stringValue() : string {
+        let image: string = "";
+        let t: Timestamp = this;
+
+        switch (t.precision) {
+            default: throw { msg: "invalid value for timestamp precision", where: "IonValueSupport.timestamp.toString" };
+            case Precision.NULL:
+                return "null.timestamp";
+            case Precision.FRACTION:
+                let digits = t.fraction.getDigits().toString();
+                //exp always negative because we are a magnitude.
+                let tempExp = t.fraction.getExponent();
+                if(digits === '0') tempExp++;
+                let zeros = ".";
+                while(tempExp < 0) {
+                    zeros = zeros + '0';
+                    tempExp++;
+                }
```

Comment:
Does `Decimal` not have a `.toString()` function?  This feels like converting a `Decimal` to a string should be in its own function, at least.

Or... maybe this is another case where `String.padLeft()` should be used.